core/aws-core/src/main/java/software/amazon/awssdk/awscore/client/handler/AwsClientHandlerUtils.java

```diff
@@ -100,6 +101,11 @@ private AwsClientHandlerUtils() {
             .putAttribute(SdkInternalExecutionAttribute.DISABLE_HOST_PREFIX_INJECTION,
                           clientConfig.option(SdkAdvancedClientOption.DISABLE_HOST_PREFIX_INJECTION));
 
+        Optional.ofNullable(clientConfig.option(SdkClientOption.EXECUTION_ATTRIBUTES)).ifPresent(clientOverrideConfiguration ->
+                executionAttributes.putAllAttributes(clientOverrideConfiguration));
+        originalRequest.overrideConfiguration().ifPresent(requestOverrideConfiguration ->
+                executionAttributes.putAllAttributes(requestOverrideConfiguration.executionAttributes()));
+
```

Comment:
Let's simplify this by using [`AttributeMap#merge()`](https://github.com/aws/aws-sdk-java-v2/blob/1df4bd27fce7f9eb8c8670b0417057b63bba0ae3/utils/src/main/java/software/amazon/awssdk/utils/AttributeMap.java#L63-L74) to merge the request level and client level execution attribute overrides, giving the request level attributes precedence.